### ISSUE TITLE
GDScript builtin functions don't work in target=release builds

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#41706]
**Godot version:**
358e209fa049887fd4212c52c9faba0ca9d6e111

**OS/device including version:**
Linux, `tools=no target=release` build with GCC 10.

**Issue description:**
When building a release template of the `master` branch (i.e. `scons p=linuxbsd tools=no target=release`), the rewritten GDScript doesn't seem to  handle builtin function calls properly.

This script:
```gdscript
func _ready():
    print("Hello")
```
Triggers:
```
SCRIPT ERROR: Parse Error: Too many arguments for "print()" call. Expected at most 0 but received 1.
          at: GDScript::reload (res://Node2D.gd:5)
ERROR: Method/function failed. Returning: ERR_PARSE_ERROR
   at: reload (modules/gdscript/gdscript.cpp:633)
```

The same happen with other builtin function calls like `var2str`.

Note: This can be a bit tricky to test due to #40601 if your Vulkan driver is radv on Linux. The current `master` branch is overall not much tested and pretty buggy in non-tools and release builds currently.

**Steps to reproduce:**
- `scons p=linuxbsd tools=no target=release`
- run on a project with a main scene's script with any call to a builtin function (at least `print` or `var2str`)

**Minimal reproduction project:**
Not necessary.

**Comment by vnen** [comment]
Ah, the problem is likely caused by the absence of method information on release. The best solution would be allowing this info to be kept in release, which would also ensure type safety when compiling it.

**Comment by akien-mga** [comment]
Seems fixed in current `master` (688228030850005f84d7b51a527bc779d76b905e).
